src/datasources/asset/AssetDataSource.ts

```diff
@@ -70,10 +96,203 @@ export class AssetDataSource extends DataSourceBase<AssetMetadataQuery> {
+    const workspaceId = this.templateSrv.replace(query.workspace);
+    let continuationToken = '';
+    const {
+      assetIdentifiers,
+      minionIds
+    } = query;
+    const workingHoursPolicy = await this.getServicePolicy();
+    let assets: string[] = replaceVariables(assetIdentifiers, this.templateSrv);
+    let systems: string[] = replaceVariables(minionIds, this.templateSrv);
+    let entityIds: string[];
+    if (query.entityType === EntityType.Asset) {
+      entityIds = assets
+    } else {
+      entityIds = systems
+    }
+    // filter empty values
+    entityIds = entityIds.filter(Boolean);
+    // retrieve the first 10 asset or system IDs to display some data, ensuring the utilization graph is not empty.
+    if (!entityIds.length) {
+      if (query.entityType === EntityType.Asset) {
+        let conditions: string[] = [];
+        if (workspaceId) {
+          conditions.push(`workspace = "${workspaceId}"`);
+        }
+        if (systems.length) {
+          const systemsCondition = systems.map(id => `${AssetFilterProperties.LocationMinionId} = "${id}"`)
+          conditions.push(`(${systemsCondition.join(' or ')})`);
+        }
+        const assetFilter = conditions.join(' and ');
+        const assetsResponse = await this.queryAssets(assetFilter, 10);
+        entityIds = assetsResponse.map((asset) => {
+          return asset.id;
+        })
+      } else {
+        let filterString = '';
+        if (query.workspace) {
+          filterString += `workspace = "${query.workspace}"`;
+        }
+        let systemMetadata = await this.getSystems({
+          filter: filterString,
+          projection: `new(${defaultProjection.join()})`,
+          orderBy: defaultOrderBy,
+          take: 10
+        })
+        entityIds = systemMetadata.data.map((system) => {
+          return system.id;
+        })
+      }
+    }
+    const resultArray: Array<{
+      id: string,
+      datetimes: number[],
+      values: number[]
+    }> = [];
+    for (let index = 0; index < entityIds.length; index++) {
+      let data: AssetUtilizationHistory[] = [];
+      let requestCount = 0;
+      do {
+        try {
+          let requestBody: QueryAssetUtilizationHistoryRequest = {
+            utilizationFilter: '',
+            continuationToken: continuationToken,
+            orderBy: AssetUtilizationOrderBy.START_TIMESTAMP,
+            orderByDescending: true
+          }
+          if (query.entityType === EntityType.Asset) {
+            requestBody.assetFilter = `${AssetFilterProperties.AssetIdentifier} = "${entityIds[index]}"`;
+          } else {
+            requestBody.assetFilter = `${AssetFilterProperties.LocationMinionId} = "${entityIds[index]}" and ${AssetFilterProperties.IsSystemController} = true`;
+          }
+          const response: AssetUtilizationHistoryResponse = await this.post<AssetUtilizationHistoryResponse>(this.baseUrl + '/query-asset-utilization-history', requestBody);
```

Comment:
The /query-asset-utilization-history seems to be a beta endpoint that we don't even list in swagger.

@TigranVardanyan @mure was this discussed with someone that we can release the Grafana plugin without properly releasing the backend endpoint ?

Reply:
We will need to do a performance test and production readiness before releasing new backend endpoints.  Let's setup a call to understand that what we are doing is actually something we can deliver for october.

Reply:
Please keep in mind we're pushing for quicker BETA release that will use an existing BETA endpoint. I don't understand the blocker of a beta release using a beta endpoint.

Reply:
@MattH-NI @TigranVardanyan  the code seems to be part of the Asset Plugin which will be included in the actual release since our team will be releasing Calibration related functionalities as well. 

Maybe I missed this but how is the user supposed to know this is only beta for SLE ? Is this feature even for SLE or only for SLS ? Also shouldn't the product team responsible for ensuring the backend can handle the load know about this ? There are various implications on the asset service that can impact / degrade performance of other asset functionalities.

It would be great to have a sync to better understand what's the plan of delivering this feature and how should we coordinate to release a common plugin that is supposed to contain both beta and new features.  

cc @TJ-G @CiprianAnton @mure

Reply:
We haven't reviewed the code for asset utilization. More than that, we know that the query-asset-utilization-history also used in asset grid can cause service instability when there is load on the service. We should avoid exposing functionality we don't know it works reliable and don't scale.

Reply:
@kkerezsi @CiprianAnton, is the ‘query-asset-utilization-history’ endpoint the only one in beta? Sorry, I can’t check right now, but in SLS there was a second one that returns utilization data, not historical data, but a single value for a range. Is that one fully tested and ready for development?

Reply:
we will mark Asset datasource plugin as _beta_